shell/platform/embedder/embedder.h

```diff
@@ -143,20 +143,39 @@ typedef struct {
   double pixel_ratio;
 } FlutterWindowMetricsEvent;
 
+// The kind of device generating a pointer event.
+typedef enum {
+  kMouse,
+  kSignal,
+} FlutterPointerDeviceKind;
+
+// The phase of the pointer event.
 typedef enum {
   kCancel,
   kUp,
   kDown,
   kMove,
+  kAdd,
+  kRemove,
+  kHover,
```

Comment:
cc @gspencergoog who has been working on hover events; please check that this is consistent with what you've been doing

Reply:
Note that this is just exposing existing values that previously existed everywhere but the embedding layer, so shouldn't conflict with anything. `kHover` will definitely be needed for hover support anyway (and the converter code wants add/remove if hover events are sent). We could move this to a separate patch now that there's already a reason to expose these independent of scroll.

Reply:
Split out into #7813